x-pack/plugins/rule_registry/server/search_strategy/search_strategy.ts

```diff
@@ -0,0 +1,142 @@
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { Logger } from 'src/core/server';
+import { from } from 'rxjs';
+import { isValidFeatureId } from '@kbn/rule-data-utils';
+import { ENHANCED_ES_SEARCH_STRATEGY } from '../../../../../src/plugins/data/common';
+import { ISearchStrategy, PluginStart } from '../../../../../src/plugins/data/server';
+import {
+  RuleRegistrySearchRequest,
+  RuleRegistrySearchResponse,
+} from '../../common/search_strategy';
+import { ReadOperations, PluginStartContract as AlertingStart } from '../../../alerting/server';
+import { SecurityPluginSetup } from '../../../security/server';
+import { SpacesPluginStart } from '../../../spaces/server';
+import { IRuleDataService } from '..';
+import { Dataset } from '../rule_data_plugin_service/index_options';
+import { MAX_ALERT_SEARCH_SIZE } from '../../common/constants';
+import { AlertAuditAction, alertAuditEvent } from '../';
+import { getSpacesFilter, getAuthzFilter } from '../lib';
+
+export const ruleRegistrySearchStrategyProvider = (
+  data: PluginStart,
+  ruleDataService: IRuleDataService,
+  alerting: AlertingStart,
+  logger: Logger,
+  security?: SecurityPluginSetup,
+  spaces?: SpacesPluginStart
+): ISearchStrategy<RuleRegistrySearchRequest, RuleRegistrySearchResponse> => {
+  const es = data.search.getSearchStrategy(ENHANCED_ES_SEARCH_STRATEGY);
+
+  return {
+    search: (request, options, deps) => {
+      const securityAuditLogger = security?.audit.asScoped(deps.request);
+      const getActiveSpace = async () => spaces?.spacesService.getActiveSpace(deps.request);
+      const getAsync = async () => {
+        const [space, authorization] = await Promise.all([
+          getActiveSpace(),
+          alerting.getAlertingAuthorizationWithRequest(deps.request),
+        ]);
+        const authzFilter = (await getAuthzFilter(
+          authorization,
+          ReadOperations.Find
+        )) as estypes.QueryDslQueryContainer;
+        return { space, authzFilter };
+      };
+      return from(getAsync()).pipe(
+        mergeMap(({ space, authzFilter }) => {
+          const indices: string[] = request.featureIds.reduce((accum: string[], featureId) => {
+            if (!isValidFeatureId(featureId)) {
+              logger.warn(
+                `Found invalid feature '${featureId}' while using rule registry search strategy. No alert data from this feature will be searched.`
+              );
+              return accum;
+            }
+
+            return [
+              ...accum,
+              ...ruleDataService
+                .findIndicesByFeature(featureId, Dataset.alerts)
+                .map((indexInfo) => {
+                  return featureId === 'siem'
+                    ? `${indexInfo.baseName}-${space?.id ?? ''}*`
+                    : `${indexInfo.baseName}*`;
+                }),
+            ];
+          }, []);
+
+          const filters = [];
```

Comment:
if we do `query?: { bool: estypes.QueryDslBoolQuery };`, we can do something like that
```
const filter = request.query?.bool?.filter ?
  Array.isArray(request.query?.bool?.filter) ?
    request.query?.bool?.filter : [request.query?.bool?.filter]
  : [];
if (authzFilter) {
  filter.push(authzFilter);
}
if (space?.id) {
  filter.push(getSpacesFilter(space.id) as estypes.QueryDslQueryContainer);
}

const query = {
  bool: {
    ...request.query?.bool,
    filter,
  },
};
const params = {
  index: indices,
  body: {
    size: MAX_ALERT_SEARCH_SIZE,
    query,
  },
};
```

Reply:
Sure!